pkg/services/rendering/http_mode.go

```diff
@@ -48,32 +49,61 @@ func (rs *RenderingService) renderViaHttp(ctx context.Context, renderKey string,
 	queryParams.Add("encoding", opts.Encoding)
 	queryParams.Add("timeout", strconv.Itoa(int(opts.Timeout.Seconds())))
 	queryParams.Add("deviceScaleFactor", fmt.Sprintf("%f", opts.DeviceScaleFactor))
+
 	rendererUrl.RawQuery = queryParams.Encode()
 
-	req, err := http.NewRequest("GET", rendererUrl.String(), nil)
+	// gives service some additional time to timeout and return possible errors.
+	reqContext, cancel := context.WithTimeout(ctx, opts.Timeout+time.Second*2)
+	defer cancel()
+
+	resp, err := rs.doRequest(rendererUrl, opts.Headers, reqContext)
 	if err != nil {
 		return nil, err
 	}
 
-	req.Header.Set("User-Agent", fmt.Sprintf("Grafana/%s", setting.BuildVersion))
+	// save response to file
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			rs.log.Warn("Failed to close response body", "err", err)
+		}
+	}()
 
-	for k, v := range opts.Headers {
-		req.Header[k] = v
+	err = rs.readFileResponse(resp, filePath, ctx)
+	if err != nil {
+		return nil, err
 	}
 
+	return &RenderResult{FilePath: filePath}, nil
+}
+
+func (rs *RenderingService) renderCSVViaHttp(ctx context.Context, renderKey string, opts CSVOpts) (*RenderCSVResult, error) {
```

Comment:
Go and initialisms in consistent case etc:
```suggestion
func (rs *RenderingService) renderCSVViaHTTP(ctx context.Context, renderKey string, opts CSVOpts) (*RenderCSVResult, error) {
```